src-tauri/src/demo/analyser/mod.rs

```diff
@@ -71,35 +72,100 @@ pub struct HighlightEvent {
     event: Highlight,
 }
 
+/// Snapshot of a player at the time a highlight occurred so highlights will display using the
+/// correct team colors.
+#[derive(Debug, Serialize, PartialEq)]
+pub struct HighlightPlayerSnapshot {
+    /// The ID of the player.
+    user_id: UserId,
+
+    /// The name of the player at the time the highlight occurred.
+    name: String,
+
+    /// What team the player was on at the time.
+    /// NOTE: This can sometimes be Team::Other (probably when an m_iTeam update is missing)
+    team: Team,
+}
+
+impl HighlightPlayerSnapshot {
+    fn for_player(player_id: u16, lookup: &HashMap<UserId, PlayerState>) -> Self {
+        let player = lookup.get(&player_id.into());
+
+        Self::for_player_with_name(player_id, player.map(|p| p.name.clone()), lookup)
+    }
```

Comment:
Maybe make this a method of `GameDetailsAnalyser` so you don't have to pass the `players` HashMap as an argument every time?